SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +12,71 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
 
         //public UserController() { }
 
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
+        public async Task<IEnumerable<User>> Get()
+        {
             var users = _context.Users.ToArray<User>();
             return users;
         }
 
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
+        public async Task<User> Get(int id)
+        {
             User user = _context.Users.Find(id);
             return user;
         }
 
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
-        }
-
         // POST: api/User
         [HttpPost]
-        public virtual void Post([FromBody] User user) {
+        public async Task<ActionResult> Post([FromBody] User user)
```

Comment:
there are no asynchronous calls being made here please fix.

Reply:
Hi, I would also like to see something in this line on your post
if (ModelState.IsValid)
            {
                // Do something with the product (not shown).

                return new HttpResponseMessage(HttpStatusCode.OK);
            }
            else
            {
                return Request.CreateErrorResponse(HttpStatusCode.BadRequest, ModelState);
            }
This is boiler plate so need love but you should get the idea